hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/diskUsage/diskUsage.tsx

```diff
@@ -144,20 +145,19 @@ export class DiskUsage extends React.Component<Record<string, object>, IDUState>
       const dataSize = duResponse.size;
       let subpaths: IDUSubpath[] = duResponse.subPaths;
 
-      subpaths.sort((a, b) => (a.size < b.size) ? 1 : -1);
-
       // Only show top n blocks with the most DU,
       // other blocks are merged as a single block
-      if (subpaths.length > limit) {
+      if (subpaths.length > limit || (subpaths.length > 0 && limit === MAX_DISPLAY_LIMIT)) {
         subpaths = subpaths.slice(0, limit);
         let topSize = 0;
-        for (let i = 0; i < limit; ++i) {
+        for (let i = 0; limit === MAX_DISPLAY_LIMIT ? i < subpaths.length : i < limit; ++i) {
```

Comment:
I think this could work wrong. What happens if the limit is set to 10, but we only have 5 subpaths? In that case we would iterate until `i` is smaller than 10? But we don't have that many elements in `subpaths`. Did I miss something?

Reply:
@dombizita 

If Limit=10 and subpath.length=5 then It will not enter in Outer If Condition
5 > 10 it will become false
if (subpaths.length > limit || (subpaths.length > 0 && limit === MAX_DISPLAY_LIMIT)) {
              subpaths = subpaths.slice(0, limit);
              let topSize = 0;
              **for (let i = 0; limit === MAX_DISPLAY_LIMIT ? i < subpaths.length : i < limit; ++i)** {
                topSize += subpaths[i].size;
              }
              const otherSize = dataSize - topSize;
              if (otherSize > 0) {
                const other: IDUSubpath = {path: OTHER_PATH_NAME, size: otherSize};
                subpaths.push(other);
              }
      }